components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -193,32 +172,24 @@ public Response getResourceIds(MessageContext context) {
     public Response updateResource(String resourceId, ResourceDetailsDTO updatedResource, MessageContext context) {
 
         try {
-            if (isValidToken(context)) {
-
-                if (log.isDebugEnabled()) {
-                    log.debug("Access token is valid and all required context information " +
-                            "are existing in the token.");
-                }
-                if (isResourceIdValid(resourceId)) {
-
+            if (isValidTokenScope(context) && isResourceIdExists(resourceId)) {
+                if (updatedResource.getResource_Scopes().isEmpty()) {
                     if (log.isDebugEnabled()) {
-                        log.debug("Valid resourceId :" + resourceId);
+                        log.debug("Cannot update resource: " + resourceId + "as resource scopes are not provided");
```

Comment:
Change to `log.debug("Cannot update resource: " + resourceId + " as resource scopes are not provided.");`

Reply:
fixed: https://github.com/wso2-extensions/identity-oauth-uma/pull/68